tests/sat_flows.rs

```diff
@@ -0,0 +1,308 @@
+        "Sending node ({}) is insufficiently funded [Outbound capacity: {outbound_capacity}, required: {payment_amount_plus_fees}]",
+        node_info.node_pubkey
+    );
+
+    tn
+}
+
+fn setup_receiver_node(max_payment_amount: u64) -> TransactingNode {
+    let tn = setup_node(Some(NodeType::Receiver));
+
+    let node_info = tn.node.get_node_info().unwrap();
+    let inbound_capacity = node_info.channels_info.inbound_capacity.sats;
+    assert!(
+        inbound_capacity > max_payment_amount,
+        "Sending node ({}) has insufficient inbound capacity: {inbound_capacity} (required: {max_payment_amount})",
+        node_info.node_pubkey
+    );
+
+    tn
+}
+
+fn setup_node(node_type: Option<NodeType>) -> TransactingNode {
+    let received_payments = Arc::new(Mutex::new(Vec::<String>::new()));
+    let sent_payments = Arc::new(Mutex::new(Vec::<String>::new()));
+
+    let before_node_started = Instant::now();
+    let node = start_specific_node(
+        node_type.clone(),
+        Box::new(ReturnFundsEventsHandler {
+            received_payments: received_payments.clone(),
+            sent_payments: sent_payments.clone(),
+        }),
+    )
+    .unwrap();
+    println!(
+        "{:?} node started in {} milliseconds",
+        node_type.unwrap(),
+        before_node_started
+            .elapsed()
+            .as_millis()
+            .separate_with_commas()
+    );
+
+    TransactingNode {
+        node,
+        received_payments,
+        sent_payments,
+    }
+}
+
+fn wait_for_payment(
+    payment_hash: &str,
+    sender: &Arc<Mutex<Vec<String>>>,
+    receiver: &Arc<Mutex<Vec<String>>>,
+) -> Result<(), &'static str> {
+    let start_time = Instant::now();
+    let mut sender_sent_payment = false;
+    let mut receiver_received_payment = false;
+    loop {
+        if start_time.elapsed().as_secs() >= MAX_PAYMENT_TIME_SECS {
+            return Err("Payment did not go through within {MAX_PAYMENT_TIME_SECS} seconds!");
+        }
+
+        if let Some(last_payment_hash) = sender.lock().unwrap().last() {
+            if last_payment_hash == payment_hash {
+                sender_sent_payment = true;
+            } else {
+                return Err("Unexpected payment sent: {last_payment_hash} (expected payment: {payment_hash})");
+            }
+        }
+
+        if let Some(last_payment_hash) = receiver.lock().unwrap().last() {
+            if last_payment_hash == payment_hash {
+                receiver_received_payment = true;
+            } else {
+                return Err("Unexpected payment received: {last_payment_hash} (expected payment: {payment_hash})");
+            }
+        }
+
+        if sender_sent_payment && receiver_received_payment {
+            return Ok(());
+        }
+
+        std::thread::sleep(std::time::Duration::from_secs(1));
+    }
+}
+
+fn get_payment_amount() -> PaymentAmount {
+    let exact = 1 + rand::random::<u64>() % MAX_PAYMENT_AMOUNT_SATS;
```

Comment:
I am unsure if there is a reason to test with different amounts, but if there is, we shouldn't use random ones (we add variability between tests). One option would be to always run 3 payments (e.g. 50, 500, 5000 sats). But always doing 5000 seems enough.

Reply:
[I agree](https://github.com/getlipa/lipa-lightning-lib/pull/1190/commits/d487ae5a052872d163b67105e882d5d29ff020ac), just not sure about the sats amount. I set it to 300 SAT to save some fees. What should we base our `golden number` on?

Reply:
For our main purpose with this, the amount shouldn't matter much. 300 is fine by me :)